pallets/services/src/lib.rs

```diff
@@ -1099,5 +1054,59 @@ pub mod module {
 			// TODO: add weight for the call to the total weight.
 			Ok(PostDispatchInfo { actual_weight: None, pays_fee: Pays::Yes })
 		}
+
+		pub fn slash(
+			origin: OriginFor<T>,
+			offender: T::AccountId,
+			#[pallet::compact] service_id: u64,
+			#[pallet::compact] percent: Percent,
+		) -> DispatchResult {
+			let caller = ensure_signed(origin)?;
+			let service = Self::services(service_id)?;
+			// TODO: add EVM hook to query for the slashing permission.
+			ensure!(service.owner == caller, DispatchError::BadOrigin);
+			let service_operator =
+				service.operators.iter().find(|(operator, _)| operator == &offender);
+			let (operator, restake_percent) = match service_operator {
```

Comment:
Is restake percent the exposed percentage of restaked assets?

Reply:
the exposed percentage.